src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -313,81 +313,34 @@ public Task<ODataBatchOperationResponseMessage> CreateOperationResponseMessageAs
 #endif
 
         /// <summary>Flushes the write buffer to the underlying stream.</summary>
-        public void Flush()
-        {
-            this.VerifyCanFlush(true);
-

```

Comment:
I don't think it would work, because the common base ODataOutputContext context doesn't have a Flush() definition. The design has been letting the derived context (raw, json or text, etc) to handle underlying writer respectively. Original BatchWriter has only one format and it can provide implementation here. Now with the additional Json format, it needs to be abstract.